pip/templates/deploy.py

```diff
@@ -29,53 +29,22 @@
 # noinspection PyUnresolvedReferences
 import twine.commands.upload
 
-parser = argparse.ArgumentParser()
-parser.add_argument('repo_type')
-args = parser.parse_args()
-
-repo_type_key = args.repo_type
-
-pip_repositories = {
-    'snapshot' : "{snapshot}",
-    'release' : "{release}"
-}
-
-pip_registry = pip_repositories[repo_type_key]
-
-pip_username, pip_password = (
-    os.getenv('DEPLOY_PIP_USERNAME'),
-    os.getenv('DEPLOY_PIP_PASSWORD'),
-)
-
-if not pip_username:
-    raise Exception(
-        'username should be passed via '
-        'DEPLOY_PIP_USERNAME env variable'
-    )
-
-if not pip_password:
-    raise Exception(
-        'password should be passed via '
-        '$DEPLOY_PIP_PASSWORD env variable'
-    )
-
 with open("{version_file}") as version_file:
     version = version_file.read().strip()
-
 new_package_file = None
-
+new_wheel_file = None
 try:
-    new_package_file = "{package_file}".replace(".tar.gz", "-{}.tar.gz".format(version))
+    os.mkdir("./dist")
+    new_package_file = "./dist/" + "{package_file}".replace(".tar.gz", "-{}.tar.gz".format(version))
+    new_wheel_file = "./dist/" + "{wheel_file}".replace(".whl", "-{}.whl".format(version))
+
     shutil.copy("{package_file}", new_package_file)
+    shutil.copy("{wheel_file}", new_wheel_file)
 
     twine.commands.upload.main([
-        new_package_file,
-        '-u',
-        pip_username,
-        '-p',
-        pip_password,
-        '--repository-url',
-        pip_registry
+        './dist/*',
+        '--repository',
+        'codeartifact'
```

Comment:
Can we remain repo agnostic?

Reply:
codeartifact is a profile in ~/.pypirc. This change is to adapt our CA config. Are you suggesting to make it a pass-in parameter? I don't see the difference because I assume we are not using other profiles for twine config

Reply:
yeah, could we have this as a parameter in the rule that would then cause the behavior to use `--repository` & `~/.pypirc` instead of the username/password?

Reply:
Ok I think I get what you said. I will add a flag to enable both behaviors.